### PR TITLE
upgrade ts-codegen and use bundles

### PR DESCRIPTION
**Pull request by pyramation**

### CODE REVIEW

**Comment by pyramation** [comment]
you may not want to use the bundle option (it automates the packaging of multiple contracts) 

the difference between the two is highlighted here https://github.com/nymlab/vectis/pull/84/commits/b5698087fe08dade8b4ba9492889b6577369e478

without bundling: https://github.com/nymlab/vectis/pull/84
with bundling: https://github.com/nymlab/vectis/pull/83

**Comment by pyramation** [comment]
Here is what the bundling looks like: 

Screen Shot 2022-08-09 at 4 11 30 PM
Screen Shot 2022-08-09 at 4 11 37 PM

File: cli/types/contracts/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import * as _4 from "./Govec.types";
+import * as _5 from "./Govec.client";
+export namespace contracts {
+  export const Proxy = { ..._0,
+    ..._1
+  };
+  export const Factory = { ..._2,
+    ..._3
+  };
+  export const Govec = { ..._4,
```
Comment (sascha1337) [inline cli/types/contracts/index.ts:20]: Spread candy 🍭

File: cli/types/package.json [hunk 2]
```diff
@@ -11,10 +11,8 @@
     "scripts": {
         "publish:dev": "tsc && cp package.json build/package.json && cd build && npm publish --dry-run",
         "publish:prod": "tsc && cp package.json build/package.json && cd build && npm publish",
-        "generate": "npm run types:proxy && npm run types:factory && npm run types:govec && npm version patch",
-        "types:proxy": "cosmwasm-ts-codegen generate --schema ../../contracts/proxy/schema --out ./contracts --name Proxy",
-        "types:factory": "cosmwasm-ts-codegen generate --schema ../../contracts/factory/schema --out ./contracts --name Factory",
-        "types:govec": "cosmwasm-ts-codegen generate --schema ../../contracts/govec/schema --out ./contracts --name Govec"
+        "generate": "npm run types && npm version patch",
```
Comment (sascha1337) [inline cli/types/package.json:14]: This is so much cleaner, exactly how I wanted to see this being implemented, good one ☝️

**Review by sascha1337 — APPROVED** [review]
Looks solid for me

**Comment by sascha1337** [comment]
@j0nl1 @whalelephant please approve pyra so we can see the CI tests spinning

**Comment by pyramation** [comment]
cool @sascha1337 glad you like it! a ton of work went in to improve CI/CD including bundling

just want to make sure you guys choose between the version of bundling I currently have, or if you want to do that manually

> the difference between the two is highlighted here [b569808](https://github.com/nymlab/vectis/commit/b5698087fe08dade8b4ba9492889b6577369e478)
> 
> without bundling: #84


I'm also down to explore making a bundling like this one that you guys did — I actually added a scope in my system, but could work to make it like this eventually: 
https://github.com/nymlab/vectis/blob/a0146d6020a44eebb49f09793546be8b4c591fd3/cli/types/index.ts

**Comment by j0nl1** [comment]
@pyramation again it is an amazing job, we preferred to use no-bundle option and do the export manually because in the bundle version the types are assigned in a const object and these types are treats as values. When we tried to type a property in a class we were getting an error saying that was a value and not a type.

**Pull request closed without merging**